core/shared/src/main/scala/fs2/internal/Algebra.scala

```diff
@@ -262,15 +262,15 @@ private[fs2] object Algebra {
           fx match {
             case output: Algebra.Output[F, X] =>
               interruptGuard(
-                F.pure(Out(output.values, scope, f(Right(()))))
+                F.pure(Out(output.values, scope, FreeC.Pure(()).transformWith(f)))
```

Comment:
@pchlupacek @AdamChlupacek Can you double check this change? I think this is necessary to preserve any error handling provided by `f`. If we just call `f(Right(()))`, we end up discarding the root level error handler -- but the exception handler in `compileScope` calls `tail.asHandler` in the event that folding the segment results in an exception.

Reply:
Good catch, we should never call `f(x)` directly, we should always use transform with it. As to the `tail.asHandler`, that seems to be there only for the case of error coming from `g`.